Show LetsEncrypt certinfo on dietpi-banner for non-root users

FrapiFrance opened the pull request:
Non-root users can't see LetsEncrypt valid end date because no access to /etc/letsencrypt/live/*
So LetsEncrypt shows "No certificate found".
This side script with sudo allows everybody to see Let's Encrypt cert status

Commit:
Show LetsEncrypt certinfo on dietpi-banner for non-root users

Commit:
fiw pointed by shellcheck

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -262,13 +262,7 @@ $GREEN_LINE"
-				certinfo=$(openssl x509 -enddate -noout -in "$i" | mawk '/notAfter=/{print "Valid until "$4"-"substr($1,10)"-"$2" "$3}' 2>&1)
-				break
-			done
+			local certinfo="$(sudo /boot/dietpi/func/dietpi-banner-certinfo)"
```
```suggestion
			Print_Cert_Info()
			{
				local i
				for i in /etc/letsencrypt/live/*/cert.pem
				do
					[ -f "$i" ] || continue
					openssl x509 -enddate -noout -in "$i" | mawk '/notAfter=/{print "Valid until "$4"-"substr($1,10)"-"$2" "$3}' 2>&1
					return 0
				done
				echo 'No certificate found'
			}
			echo -en "$GREEN_BULLET ${aCOLOUR[1]}${aDESCRIPTION[16]} "
			if (( $UID ))
			then
				sudo dash -c "$(declare -f Print_Cert_Info); Print_Cert_Info"
			else
				Print_Cert_Info
			fi
```
That function declaration should probably be done above the `Print_Banner` function, like `Print_Local_Ip` etc. I used `dash` (bourne shell) instead of `bash`, which is faster. I hope that `echo -en` (no line break) works in combination with a 2nd `echo` from within `sudo` session, to be shown in one line 🙂.

Of course, the banner would be broken anyway, when that `sudo` call asked for a password. Not sure whether there is a quick/simple way to check whether the current user has `NOPASSWD` `sudo` permissions or not. If there is a way, I would prefer to just skip the cert info then, to not have a password prompt on every login, right after that has been entered for login, in many cases.

FrapiFrance replied:
Oh, I did quite the same without seeing your suggestion ; I update with dash, and NOPASSWD control

FrapiFrance replied:
Actually,
Won't use dash.

With dash `[[ -e "$i" ]]` won't work
If I use [ -e "$i" ], shellcheck won't accept : 
```
In ./dietpi/func/dietpi-banner line 227:
                [ -e "$i" ] || continue
                ^---------^ SC2292 (style): Prefer [[ ]] over [ ] for tests in Bash/Ksh.

Did you mean: 
                [[ -e "$i" ]] || continue

For more information:
  https://www.shellcheck.net/wiki/SC2292 -- Prefer [[ ]] over [ ] for tests i...
Error: Process completed with exit code 1.
```
=> I offer to use bash, to keep the script shellcheck-compliant

FrapiFrance replied:
Should be OK now, ready for review

MichaIng replied:
You can tell shellcheck to ignore issues. It is a help to find possible issues and help with consistent coding style, not something that should prevent us from more efficient code 😉.

I'll apply some refinement.

Commit:
sudo only when needed ; no external script

Commit:
dash-ccompatible Print_Cert_Info

Commit:
better test

Commit:
trailing spaces

Commit:
shellchack is more resstrictive than my raspberry ;-)

Commit:
[[ ]] in the BASH-part of the script

Commit:
shellcheck-compliant more important than dash

Commit:
last test with dash

Commit:
definitely bash

MichaIng committed:
Minor coding enhancements

Use dash instead of bash for printing LE cert status via sudo, merge NOPASSWD check and actual sudo call into one, and avoid command substitutions where possible.
